src/LtiLibrary.NetCore/Clients/SecuredClient.cs

```diff
@@ -17,46 +17,47 @@ public static class SecuredClient
         /// <summary>
         /// Add a signed authorization header to the client request using the signatureMethod.
         /// </summary>
-        /// <param name="client">The HttpClient that will make the request.</param>
-        /// <param name="method">The HttpMethod of the request.</param>
-        /// <param name="serviceUrl">The service URL.</param>
-        /// <param name="content">The Content of the request (used to calculate the body hash).</param>
+        /// <param name="client">May need BaseAddress value from this</param>
+        /// <param name="request">The request object that will be sent</param>
         /// <param name="consumerKey">The OAuth consumer key.</param>
         /// <param name="consumerSecret">The OAuth consumer secret.</param>
         /// <param name="signatureMethod">The SignatureMethod used to sign the request.</param>
         /// <returns></returns>
-        public static async Task SignRequest(HttpClient client, HttpMethod method, string serviceUrl,
-            HttpContent content, string consumerKey, string consumerSecret, SignatureMethod signatureMethod)
+        public static async Task SignRequest(HttpClient client, HttpRequestMessage request, string consumerKey, string consumerSecret, SignatureMethod signatureMethod)
         {
             if (client == null)
             {
-                throw new ArgumentNullException(nameof(client));
+               throw new ArgumentNullException(nameof(client));
             }
 
-            if (string.IsNullOrEmpty(serviceUrl))
+            if (request == null)
             {
-                throw new ArgumentNullException(nameof(serviceUrl));
+                throw new ArgumentNullException(nameof(request));
             }
 
-            var serviceUri = new Uri(serviceUrl, UriKind.RelativeOrAbsolute);
-            if (!serviceUri.IsAbsoluteUri)
+            string serviceUrl = request.RequestUri.OriginalString;
+
+            if (!request.RequestUri.IsAbsoluteUri)
             {
-                if (client.BaseAddress == null)
+                if (!string.IsNullOrWhiteSpace(client.BaseAddress.PathAndQuery) && client.BaseAddress.PathAndQuery != "/")
                 {
-                    throw new LtiException("If serviceUrl is relative, client.BaseAddress cannot be null.");
-                }
 
-                if (!Uri.TryCreate(client.BaseAddress, serviceUrl, out serviceUri))
+                    // This is trying to replicate the behavior of httpclient combining base address with a relative path - 
+                    // if a path is in the base address, it MUST have a trailing slash, and the relative url must NOT,
+                    // so this simple combination should work
+                    serviceUrl = client.BaseAddress.AbsoluteUri + serviceUrl;
+                }
+                else
                 {
-                    throw new LtiException($"Cannot form a valid URI from {client.BaseAddress} and {serviceUrl}.");
```

Comment:
is there a situation where we should throw similarly here?  like in the helper I just wrote I checked that the 'request' object was !IsAbsoluteUrl because it didn't seem to make sense for client to have a base URL configured and the request to go against an entire different hostname?

Reply:
This only happens in a block of code that is guarded by whether the request `IsAbsoluteUri` on line 40.  Is that the check you mean, or am I missing something?